splink/find_brs_with_comparison_counts_below_threshold.py

```diff
@@ -64,30 +69,13 @@ def _generate_blocking_rule(
     """Generate a Splink blocking rule given a list of column names which
     are provided as as string"""
 
-    # TODO: Refactor in Splink4
-    dialect = linker._sql_dialect
-
-    module_mapping = {
-        "presto": "splink.athena.blocking_rule_library",
-        "duckdb": "splink.duckdb.blocking_rule_library",
-        "postgres": "splink.postgres.blocking_rule_library",
-        "spark": "splink.spark.blocking_rule_library",
-        "sqlite": "splink.sqlite.blocking_rule_library",
-    }
-
-    if dialect not in module_mapping:
-        raise ValueError(f"Unsupported SQL dialect: {dialect}")
-
-    module_name = module_mapping[dialect]
-    block_on_module = __import__(module_name, fromlist=["block_on"])
-    block_on = block_on_module.block_on
-
     if len(cols_as_string) == 0:
-        return block_on("1")
+        br: BlockingRuleCreator = CustomRule("1=1", linker._sql_dialect)
```

Comment:
This type hint was needed for mypy, without it you got an `splink/find_brs_with_comparison_counts_below_threshold.py:75: error: Incompatible types in assignment (expression has type "BlockingRuleCreator", variable has type "CustomRule")  [assignment]` error